File: JS/283-move-zeroes.js
// two pointers
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let lo = 0;
    for (let hi = 0; hi < nums.length; hi++) {
        if (nums[hi] !== 0) {
            nums[lo] = nums[hi];
            if (lo != hi) nums[hi] = 0;
            lo++;
        }
    }
};

File: C++/283-move-zeroes.cpp
#include "header.h"

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int slowIndex = 0;
        for (int fastIndex = 0; fastIndex < nums.size(); fastIndex++) {          
            if (nums[fastIndex] != 0) {
                nums[slowIndex] = nums[fastIndex];
                if (slowIndex != fastIndex) {
                    nums[fastIndex] = 0;
                }
                slowIndex++;
            }             
        }
    }
};